pandas/io/json/json.py

```diff
@@ -320,49 +334,140 @@ def read_json(path_or_buf=None, orient=None, typ='frame', dtype=True,
+
+    If initialized with ``lines=True`` and ``chunksize``, can be iterated over
+    ``chunksize`` lines at a time.
+    """
+    def __init__(
+        self, filepath_or_buffer, orient, typ, dtype, convert_axes,
+        convert_dates, keep_default_dates, numpy, precise_float, date_unit,
+        encoding, lines, chunksize, raw_json=False
+    ):
+
+        self.path_or_buf = filepath_or_buffer
+        self.orient = orient
+        self.typ = typ
+        self.dtype = dtype
+        self.convert_axes = convert_axes
+        self.convert_dates = convert_dates
+        self.keep_default_dates = keep_default_dates
+        self.numpy = numpy
+        self.precise_float = precise_float
+        self.date_unit = date_unit
+        self.encoding = encoding
+        self.lines = lines
+        self.chunksize = chunksize
+        self.nrows_seen = 0
+        self.raw_json = False
+
+        if isinstance(filepath_or_buffer, compat.string_types):
+            try:
+                exists = os.path.exists(filepath_or_buffer)
+
+            # if the filepath is too long will raise here
+            # 5874
+            except (TypeError, ValueError):
+                exists = False
+
+            if exists:
+                self.data, _ = _get_handle(filepath_or_buffer, 'r',
+                                           encoding=encoding)
+            else:
+                self.raw_json = True
+                self.data = filepath_or_buffer
+        elif hasattr(filepath_or_buffer, 'read'):
+                self.data = filepath_or_buffer
+        else:
+            self.raw_json = True
+            self.data = filepath_or_buffer
+
+        if self.raw_json and lines:
+            self.data = self.combine_lines(self.data)
+
+    def combine_lines(self, data):
+        """Combines a multi-line JSON document into a single document"""
         # If given a json lines file, we break the string into lines, add
         # commas and put it in a json list to make a valid json object.
-        lines = list(StringIO(json.strip()))
-        json = '[' + ','.join(lines) + ']'
-
-    obj = None
-    if typ == 'frame':
-        obj = FrameParser(json, orient, dtype, convert_axes, convert_dates,
-                          keep_default_dates, numpy, precise_float,
-                          date_unit).parse()
-
-    if typ == 'series' or obj is None:
-        if not isinstance(dtype, bool):
-            dtype = dict(data=dtype)
-        obj = SeriesParser(json, orient, dtype, convert_axes, convert_dates,
-                           keep_default_dates, numpy, precise_float,
-                           date_unit).parse()
-
-    return obj
+        lines = list(StringIO(data.strip()))
+        return '[' + ','.join(lines) + ']'
+
+    def read(self):
+        """Read the whole JSON input into a pandas object"""
+        if self.raw_json:
+            return self._get_obj(self.data)
+        elif self.lines and self.chunksize:
+            return concat(self)
+        else:
```

Comment:
don't need the else

Reply:
Moved this around a little bit to only have one return statement, so we can call `self.close()` in only one place. The else is now needed.